webClient/src/app/core/menu-bar/menu-bar.component.ts

```diff
@@ -567,19 +567,51 @@ export class MenuBarComponent implements OnInit, OnDestroy {
     this.editorControl.toggleTree.next();
   }
 
-  closeAll() {
-    let closeAllRef;
-    if (this.fileCount == 0) { //TODO: Enhance such that closeAll not visible if no tabs are open
+  promptToSave(file: ProjectContext): Promise<any>{
+    return new Promise((resolve, reject) => {
+      if(file.changed) {
+        const title = 'Do you want to save the changes you made to \'' + file.name + '\?';
+        const warningMessage = 'Your changes will be lost if you don\'t save them.';
```

Comment:
*don't

Reply:
I have used \ to tell the editor that ' is character and not end of string.